### DISCUSSION TITLE
"Calibration accepted" but only small effect @G6 sensor restart - multiple attempts needed

### DISCUSSION THREAD

**Discussion by emp-00** [NightscoutFoundation/xDrip#2209] [Q&A]
Have recently **updated from xDrip+ 2022-03-11 to 2022-06-20**. I've been calibrating "restarted G6 sensors" many, many times without this issue: Calibrating "down" as needed when restarting G6 does not work properly any more. The transmitter accepts calibrations but only calibrates **approx. 1/3** of the way down. 

See screenshot below: At 1st attempt reading was 202 and I calibrated to 105 -> xDrip only got it down to 176 ! Next attempt 10 mins later had "no effect at all" (probably calibrating too fast? but still, transmitter in log saying "calibration accepted" which is obviously untrue). I already had the same effect 10 days ago - it took me approx. 5 calibrations in 2 hours to finally bring readings down to the right level. In the past with the older nightlies **I always calibrated "1-step" down** without an issue. I never needed multiple attempts... Has anything changed in xDrip+ calibration code or is this caused by a new transmitter firmware? I currently have an "8U" tx rev. 2.27.2.108. Thanks for any hints.

By the way: Error message "Received backfill packet out of sequence" in the logs I think is normal - at least this also always appeared in the older version. Not sure what this means, does anybody know? Is this relevant at all?

![image](https://user-images.githubusercontent.com/60714349/177414705-571609d4-daca-4416-a66c-8401c142eb75.png)
![image](https://user-images.githubusercontent.com/60714349/177414575-f5d55e7b-1518-410a-8db2-2373eb95dfaa.png)

**Comment by Navid200** [comment]
Would you please post a screenshot of the system status page so that we can see the firmware number?  If it is really 2.27.2.108, we will need to update xDrip.
You can blackout the transmitter ID if you like.

**Reply by emp-00** [reply]
Thanks Navid200 for your help. Screenshot below - sorry, it was a typo, actually it's 2.27.2.**103**. So probably not a new firmware, right? Why am I seeing these annoying calibration issues? Last calibration attempt now 1 hour 10 minutes after the first calibration -> as you can see: 4 calibrations having absolutely no effect despite event log always confirming "Calibration accepted by transmitter". Still always getting "Received backfill packet out of sequence" errors. Calibration seems broken! All I can do is wait and try again ... :-(

![image](https://user-images.githubusercontent.com/60714349/177420229-1e29012f-3239-4589-bfce-548524f83332.png)

**Reply by Navid200** [reply]
There has been no change to calibration.
For this transmitter, calibration is done by the transmitter.
I am surprised if this is new to you.  I don't understand how this could have been any different before.

This is my experience with calibrating a G6 and its limitations and it's been like that for a very long time:  https://navid200.github.io/xDrip/docs/Calibrate-G6.html

This is my instructions for calibrating after a restart:  https://navid200.github.io/xDrip/docs/Calibrate-after-G6Restart.html

Yes, those logs (out of sequence) are expected.  It is on my todo list to remove them from the logs.

**Reply by emp-00** [reply]
I know your FAQ and your calibration instructions - and I always wondered why this was written so carefully. I can 100% confirm that I calibrated many, many sensor restarts with "1-step" down exactly as in this attempt. First time it fails for me!

I just checked under "Less common settings" / "Advanced Calibration": If I remember correctly in the past there were additional options. Also I remember, that I changed some setting in this sub-menu which may have been the reason for this to work. However, now I am unable to find these settings - looks like this has been changed?

**Reply by Navid200** [reply]
I have made changes to the menu.
For example, if you enable automatic calibration, it will ignore any settings you have chosen for entering a blood glucose reading through the treatment menu and then asking to use it for calibration.  it would be pointless if you have automatic calibration on.

But, why would you enable automatic calibration for a Dexcom transmitter in native mode? 

I also don't understand why you have engineering mode on.

If you can tell me how you can calibrate in one shot with a previous version of xDrip, it will be great.
It is possible that I have overlooked a capability of these transmitters.

**Reply by Navid200** [reply]
What I mean by that is to please tell me what version of xDrip lets you do that with this same transmitter.  And please provide the settings required to accomplish it.

**Reply by emp-00** [reply]
Exactly - there was an option "AUTO calibration" when entering BG readings. This is the option I had enabled ~2 years ago when I was still using an older 80x transmitter. After that I never touched these settings and calibrated the 8G+ Firefly's successfully with "1-step". I'm 100% sure!

**Reply by Navid200** [reply]
Please have a look at this image:
![Screenshot_20220705-175405](https://user-images.githubusercontent.com/51497406/177423328-fae58b23-2aa4-4129-9693-df52a1f0fc04.png)

Do you mean you enable 1 or 2?

**Reply by emp-00** [reply]
I just found an older "Issues conversation" where I documented the change I made in May 2020: https://github.com/NightscoutFoundation/xDrip/issues/1310#issuecomment-626937900

**Reply by emp-00** [reply]
Yep regarding your image - I had enabled "1" in the past... However, I my current nightly I cannot see these settings any more, see screenshot -- now 6 unsuccessful calibration attempts (!!) ... I tried to calibrate closer to the current reading as recommended in your FAQ but still no success :-(

![image](https://user-images.githubusercontent.com/60714349/177424667-1f13ba49-ed3a-4551-a52a-51809212dd2e.png)

**Reply by Navid200** [reply]
If you go back to the May 19, 2022 release, you will be able to enable automatic calibration and also enable the treatment menu option you like.

Would you please do that and let me know if that let's you calibrate in one shot?

**Reply by emp-00** [reply]
Sorry, just found the setting in the sub-menu of "Use Treatment BG values", Here I have set "Automatic mode (best)".

**Reply by emp-00** [reply]
I will take me approx. 30mins to revert to the May 19 release - I will try and let you know how it works!

**Reply by Navid200** [reply]
Thanks for doing the test.
So, you have automatic calibration enabled and the treatment menu set to automatic.  
Now, you use the treatment menu to calibrate?  
Or, do you use the treatment menu to enter a blood glucose reading, then, tap on it and ask it to be used for calibration.
Or, do you do something other than those two?

**Reply by emp-00** [reply]
FYI Here's an example of a successful "1-shot calibration" found in my History - calibrating from 195 to 135 mg/dl at that time. The white dots are deleted readings (just to remove those invalid values from my statistics).

![image](https://user-images.githubusercontent.com/60714349/177429323-b8838bc8-ede7-4cb7-ab4f-2d8ec72b6f99.png)

**Reply by emp-00** [reply]
I always calibrate by entering a BG value in the "syringe menu" -- due to the AUTO calibration setting this is used as calibration (without any additional prompt).

**Reply by Navid200** [reply]
Thanks
I will test as well in a few days.

**Reply by Navid200** [reply]
If you enable automatic calibration, and enter a blood glucose reading in the treatment menu, it will be used for calibration if the trend is flat since automatic calibration is enabled.
That has not changed.

**Reply by emp-00** [reply]
Now I have reverted to "my" last known good version **2022-03-11** which I used before upgrading to 2022-06-20. However, alas - it does not work any more! I have a flat BG trend and need to calibrate down from the ~170's to 110 and I even tried 125 twice --> calibration is now completely broken?! 3 hours after restart and nothing works?! I'm completely stuck now. :-(

Those "Received backfill" messages have now disappeared though.

![image](https://user-images.githubusercontent.com/60714349/177430827-e6919e47-cae0-4b4f-a8b4-86c598eebc72.png)
![image](https://user-images.githubusercontent.com/60714349/177430928-da7ab1b6-2ea3-4006-b5b3-fe923a7ef6ae.png)

**Reply by Navid200** [reply]
What was the transmitter firmware version that this methodology worked on?

At this point, if you follow my instructions, you should be able to correct the readings even though it may take a while.

**Reply by emp-00** [reply]
I have always been using Firefly's as supplied by Dexcom - successful "1-shot calibrations" have always worked for me until today!

**Comment by emp-00** [comment]
SUCCESS --- finally!! :-) Calibrating -49 mg/dl (maybe 50 delta is the limit?!)! Still running on 2022-03-11. 

I checked my History and must admit, that I don't have many examples where I had over 50 delta after a restart. This seems to be an extreme case with an extreme "restart delta". Usually I had always ~30-40 mg/dl delta values which I always successfully calibrated with the "1-shot calibration" in the past. I never calibrated twice, honestly. By the way, this "confused calibration" error you're mentioning in the FAQ I only remember from the days with the old 80x transmitters. Have never seen this error with a Firefly.

![image](https://user-images.githubusercontent.com/60714349/177431618-9bd55c6c-f18d-4622-89b4-b9e663d7e0e9.png)

**Reply by Navid200** [reply]
If I can see success with a calibration correction greater than 20% with a particular setting, I will update my guide and if I have removed the setting, I will discuss with the developers and try to get their approval to bring the setting back.

Without that, I have no choice but to continue to ask people to use instructions that provide predictable results for everyone.

With respect to the change that I made, there is a menu that let's user decide how a blood glucose entry should be used.
If you select always, it will always uses the entry for calibration regardless of the trend.
If never is selected, the entry will not be used for calibration even if the trend is flat.  If ask me every time is selected, the user will be asked if he/she wants to use the entry for calibration or not.
And if auto is selected, the user is asked if he wants to enable automatic calibration or not.

But, if you enable automatic calibration, all of those options become nonsense!
Automatic calibration does not calibrate for 10 minutes but monitors the trend.  If the trend remains flat for 10 minutes, it sends the calibration with a time parameter of 10 minutes ago.
So, all of the options in the previous paragraph become contradictory with having automatic calibration enabled.

The change I have made is to skip the options if automatic calibration is enabled.  The priority is given to automatic calibration.

**Reply by emp-00** [reply]
Regarding your comment on the 20% limit:
- see screenshot above: I successfully calibrated from 184 down to 135 mg/dl - that's **27%** and thus >20%, right? I suspect the limit is maybe rather 30% or it could be max 50 mg/dl absolute delta (?)
- successful 1-step calibration done with xDrip+ nightly 2022-03-11, if you need me to upload my complete set of settings let me know
- my experience with **nightly 2022-06-20**, see above; I'm quite sure that something has changed after the 2022-06-20 revision because calibration behavior has changed significantly for me (see my description above)

Regarding your comments on the change of the calibration settings:
- my successful setting was 100% sure and currently still is always with the "Automatic mode (best)" setting ON
- additionally I have checked "Automatic Calibration" ON (setting below Use Treatment BG values)
- last but not least I have "Infrequent Calibration" ON, but probably this was/is default as the other settings in Advanced Calibration

One more thing which could really be improved:
- my various calibration attempts yesterday have **always** been acknowledged with "**Calibration accepted**"
- obviously this is not correct; 95% of my attempts have either been rejected by the transmitter or by the xDrip algorithm (if I understand your explanation correctly, with the "Automatic mode" setting xDrip checks if the trend is "flat" and only sends the calibration to the tx if this check is passed; however the user does not get any feedback if this check has passed and/or if the tx rejects the calibrations
- in your FAQ you mention a waiting time of 25 mins for the 1st and even 45 mins for the 2nd "re-calibration": this means probably many of my calibration attempts have been rejected by the tx due to an internal timer check, correct? It would be great if the user got this as feedback e.g. "please wait XX more minutes until the next calibration attempt" either in the user interface or at least in the Event Log. In my case I was completely "blind", only seeing "accepted" calibrations which obviously have not been accepted either due to too high delta (>30%?) or insufficient waiting time; does the tx not send an error code? if it does, it looks as if xDrip does not interpret the reply code(s) correctly. Currently xDrip+ obviously always believed that calibrations are "accepted" which is not correct.

I tried to be as clear as possible, hopefully it makes sense to you and others? Let me know if I can test/try anything in order to improve the situation. G6 calibration after restarts is giving me headaches recently but other than that I'm a very happy xDrip+ user and very grateful to your (Navid200) and other developers efforts. Keep up the great work!

**Reply by Navid200** [reply]
What is in my documentation is meant to help all users.
I didn't want to state use a 25% correction limit and if it fails, reduce it to 20%.  I have experienced confused calibration after an attempt of 25% correction.  I have never experienced a confused calibration after a 20% attempt.  There is a safety margin and I wanted it that way.  I want to make sure if someone follows my guide, they will never experience confused calibration.

Yes, it is possible to attempt something that is more than 20% and get away with it.  But, if I state 25% in my documentation and only one person experiences a problem, I will still consider that a failure on my part.
In other words, my guide is intentionally conservative.

The 20% figure is only something I have found by trial and error.
It is quite possible that it is not that simple and it may be a combination of a constant and a percentage depending on the absolute level.  But, again, I have intentionally wanted it to be as simple as possible and guaranteed to work.

The same goes for the delay between consecutive calibrations.  
I did get a confused calibration when I attempted a second cal shortly after the first.  

My restart guide is based on trial and error, not xDrip code.

I have no doubt that if after a restart your error is less than 20%, you can calibrate in one shot.
But, I have never had an error less than 20% after a restart.  
I don't want a guide that helps most but not all.  I want a guide that helps all.

Restarting a G6 sensor interferes with factory calibration.  G5 does not have factory calibration.  We have to be careful what the documentation states to avoid causing harm to individuals who come from G5 to G6 and expect the same behavior.  That's why the first sentence in my guide reads "This can be very dangerous" in red.  And it goes to explain the error after a restart.
We have to be careful how everyone is going to be affected by changes we make to xDrip, not just the experts.

I believe the message "calibration accepted" from the transmitter means that the transmitter has received the calibration entry.  It doesn't necessarily mean any more than that.  But, that's just a guess.

The only thing that would help is if you could show how one can recreate what you claim.  
For example, state:
If you use this version of xDrip and if you set xDrip settings as shown here, you can accomplish a correction of 40% all the time with a single calibration.
And if you keep all the settings but change the xDrip version to the latest, it will fail.
If a developer can recreate a problem, it will go a long way.

**Reply by emp-00** [reply]
I agree it makes sense that your FAQ wording is "conservative". Understood & appreciated.

Regarding your comment "_I have experienced **confused calibration**_" with both >20% deviation and too fast re-calibration attempts: Please see my documented experience above with even 210 down to 110 ml/dl calibration (= more than 50% deviation) and attempts every ~10 to 20 minutes. I **never** got a single "confused calibration" error, the session kept running and I checked the Event Log for sure. This really never happened for me in the last 2 years with all Firefly's and many, many successful "1-shot calibrations" after restarts. I believed until today that this error was "history" from the old 80x transmitters and Native mode operations. I have also not seen other people in various forums mentioning this error any more. Are you 100% sure this still happens? Can you reproduce it? I did "my best" yesterday, luckily no success - right?

I will continue to observe the situation and with my next restart I will again revert to nightly 2022-03-11 trying to reproduce the behavior I have enjoyed for the last 2 years. I am still convinced that something has changed after this revision as documented above but I agree, cannot fully prove yet. Will do my best in the coming weeks - but it will take some time since restarts of course cannot be done very often...

**Comment by Navid200** [comment]
Since many follow my guideline, confused calibration isn't coming up a lot.
But, it still does occasionally.
This is from a post on facebook today.  It is with xDrip version of April 19, 2022 and it is with an 8L.  So, definitely not an old G6.
![296376958_10159068496306242_4503161054644007254_n](https://user-images.githubusercontent.com/51497406/182024241-fb993254-9570-4305-844c-4003f3d03945.jpg)
:
